File: Keyword.cpp
#include <iostream>
#include <cstdlib>

using namespace std;

string keywordPicker(){
    cout << "Keyword must contain each letter A-Z exactly once, no repetition of letter should occur" << endl;
    string keywrd;
    cout << "Enter keyword: ";
    getline(cin, keywrd);
    cin.ignore();

    //keyword to uppercase
    for(int i = 0; i < keywrd.length(); i++){
        keywrd[i] = toupper(keywrd[i]);
    }

    //checking repetition of letter in keyword
    for (int i = 0; i < keywrd.length(); i++){
        int pos = i;
        for(int j = 0; j < keywrd.length(); j++){
            if(pos == j){
                continue;
            } else if(keywrd[i] == keywrd[j]){
                cout << "Letter \"" << keywrd[i] << "\" Repeating in keyword" << endl;
                exit(EXIT_FAILURE); // terminating the program
            }
        } //inner for
    } // for

    //generating alphabet with keyword
    string temp = "";
    for(int i = 0; i < keywrd.length(); i++){
        temp += keywrd[i];
    }
    for(int i = 0; i < 26; i++){
        temp += (char)(i+65);
    }

    //removing letters of keyword from alphabet
    for (int i = 0; i < temp.length(); i++){
        bool found = false;
        for(int j = 0; j < keywrd.length(); j++){
            if(temp[i] == keywrd[j]){
                found = true;
                break; // no need to iterate any further
            }
        } // inner for
        if(found == false){
            keywrd += temp[i];
        }
    } // for loop

//    cout << keywrd;

    return keywrd;
}

void cipherEncryption(string alpha, string keywrd){
    string message;
    cout << "Enter message: ";
    getline(cin, message);
    cin.ignore();

    //message to uppercase
    for(int i = 0; i < message.length(); i++){
        message[i] = toupper(message[i]);
    }

    string encryText = "";
    for (int i = 0; i < message.length(); i++){
        if(message[i] == (char)32){
            encryText += " ";
        } else {
            int counter = 0;
            for (int j = 0; j < alpha.length(); j++){
                if(message[i] == alpha[j]){
                    //character at count location of alphabet is encrypted letter in key
                    encryText += keywrd[counter];
                    break;
                } else {
                    counter++;
                }
            } // inner for
        } //if-else
    } //for

    cout << "Encrypted Text: " << encryText;
}

void cipherDecryption(string alpha, string keywrd){
    string message;
    cout << "Enter Encrypted Message: ";
    getline(cin, message);
    cin.ignore();

    //message to uppercase
    for(int i = 0; i < message.length(); i++){
        message[i] = toupper(message[i]);
    }

    string decryText = "";
    for (int i = 0; i < message.length(); i++){
        if(message[i] == (char)32){
            decryText += " ";
        } else {
            int counter = 0;
            for (int j = 0; j < alpha.length(); j++){
                if(message[i] == keywrd[j]){
                    //character at count location of alphabet is encrypted letter in key
                    decryText += alpha[counter];
                    break;
                } else {
                    counter++;
                }
            } // inner for
        } //if-else
    } //for

    cout << "Decrypted Text: " << decryText;
}

int main()
{
    int choice;
    cout << "1. Encryption\n2. Decryption\nChoose(1,2): ";
    cin >> choice;
    cin.ignore();

    string alpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    string keywrd = keywordPicker();

    if(choice == 1){
        cout << endl << "---Encryption---" << endl;
        cipherEncryption(alpha, keywrd);

    } else if (choice == 2){
        cout << endl << "---Decryption---" << endl;
        cipherDecryption(alpha, keywrd);

    } else {
        cout << endl << "Wrong choice" << endl;
    }

    return 0;
}

File: Keyword.py
import sys
def keyword_picker():
    print("Keyword must contain each letter of A-Z exactly once, no letter should repeat itself")
    keyword = input("Enter keyword: ").upper()
    for i in range(len(keyword)):
        pos = i
        for j in range(len(keyword)):
            if pos == j:
                # skipping letter comparison with itself
                continue
            elif keyword[i] == keyword[j]:
                print("Letter {} repeating in keyword".format(keyword[i]))
                sys.exit()
            # elif
        # inner for
    # for

    temp = ""
    for i in range(len(keyword)):
        temp += keyword[i]
    for i in range(26):
        temp += chr(i+65)

    # removing keyword letters from alphabet
    alpha_with_key = ""
    for i in range(len(temp)):
        found = False
        for j in range(len(alpha_with_key)):
            if temp[i] == alpha_with_key[j]:
                found = True
                break
            # if
        # inner for
        if not found:
            alpha_with_key += temp[i]
        # if
    # for

    # print(alpha_with_key)

    return alpha_with_key


def encryption(alpha_key, alpha):
    message = input("Enter message: ").upper()

    encrypted_text = ""
    for i in range(len(message)):
        if message[i] == chr(32):
            encrypted_text += " "
        else:
            counter = 0
            for j in range(len(alpha)):
                if message[i] == alpha[j]:
                    encrypted_text += alpha_key[counter]
                    break
                else:
                    counter += 1
            # inner for
        # if-else
    # for

    print("Encrypted Message: {}".format(encrypted_text))


def decryption(alpha_key, alpha):
    message = input("Enter Encrypted Message: ").upper()

    decrypted_text = ""
    for i in range(len(message)):
        if message[i] == chr(32):
            decrypted_text += " "
        else:
            counter = 0
            for j in range(len(alpha)):
                if message[i] == alpha_key[j]:
                    decrypted_text += alpha[counter]
                    break
                else:
                    counter += 1
            # inner for
        # if-else
    # for

    print("Decrypted Text: {}".format(decrypted_text))


def main():
    alpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    alpha_key = keyword_picker()

    choice = int(input("1. Encryption\n2. Decryption\nChoose(1,2): "))
    if choice == 1:
        print("---Encryption---")
        encryption(alpha_key, alpha)

    elif choice == 2:
        print("---Decryption---")
        decryption(alpha_key, alpha)

    else:
        print("Incorrect Choice")

if __name__ == "__main__":
    main()
